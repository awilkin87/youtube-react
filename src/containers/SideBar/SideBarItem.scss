@import '../../styles/shared.scss';

.sidebar-item {
  i.icon {
    margin-right: 15px;
    color: #888888;
  }

  &.highlight {
    span {
      font-weight: 600;
    }

    i.icon {
      color: red;
    }
  }
}

.sidebar-item-alignment-container {
  display: flex;
  align-items: center;
}