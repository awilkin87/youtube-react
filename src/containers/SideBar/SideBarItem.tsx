import { Icon, SemanticICONS } from 'semantic-ui-react';
import SideBarClickableItem from './SideBarClickableItem';
import styles from './SideBarItem.module.scss';

/** React props for SideBarItem. */
export interface SideBarItemProps {
  highlight?: boolean,
  icon: SemanticICONS,
  label: string
}

/** Clickable sidebar menu item with a static icon and label. */
const SideBarItem: React.FunctionComponent<Readonly<SideBarItemProps>> = function (props) {
  const highlightStyle = props.highlight ? styles.highlight : null;
  return (
    <SideBarClickableItem className={[styles.sidebarItem, highlightStyle].join(' ')}>
      <div className={styles.alignmentContainer}>
        <span><Icon size='large' name={props.icon} /></span>
        <span>{props.label}</span>
      </div>
    </SideBarClickableItem>
  );
}

export default SideBarItem;