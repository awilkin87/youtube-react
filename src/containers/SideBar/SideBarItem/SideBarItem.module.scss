@use '../sidebar-common.modules';

.sidebarItem {
  :global(i.icon) {
    margin-right: 15px;
    color: sidebar-common.$grey;
  }

  &.highlight {
    span {
      font-weight: 600;
    }

    // Icon class added by Semantic UI 
    :global(i.icon) {
      color: sidebar-common.$red;
    }
  }
}

.alignmentContainer {
  display: flex;
  align-items: center;
}