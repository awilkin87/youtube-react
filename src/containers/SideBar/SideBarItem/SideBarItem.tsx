import { Icon, SemanticICONS } from 'semantic-ui-react';
import SideBarClickableItem from '../SideBarClickableItem/SideBarClickableItem';
import styles from './SideBarItem.module.scss';

/** React props for SideBarItem. */
export interface SideBarItemProps {
  /** True if this menu item should be emphasized. */
  highlight?: boolean,
  /** Semantic UI icon name. */
  icon: SemanticICONS,
  /** Menu item text. */
  label: string
}

/** Clickable sidebar menu item with a static icon and label. */
const SideBarItem: React.FunctionComponent<Readonly<SideBarItemProps>> = function (props) {
  const highlightStyle = props.highlight ? styles.highlight : null;
  return (
    <SideBarClickableItem className={[styles.sidebarItem, highlightStyle].join(' ')}>
      <div className={styles.alignmentContainer}>
        <span><Icon size='large' name={props.icon} /></span>
        <span>{props.label}</span>
      </div>
    </SideBarClickableItem>
  );
}

export default SideBarItem;