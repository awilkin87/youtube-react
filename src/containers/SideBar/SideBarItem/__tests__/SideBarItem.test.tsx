import { shallow } from "enzyme";
import SideBarItem from '../SideBarItem';

describe('SideBarItem', () => {
  test('Renders SideBarItem w/ Highlight', () => {
    const wrapper = shallow(<SideBarItem highlight icon='home' label='Menu Label' />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Renders SideBarItem w/o Highlight', () => {
    const wrapper = shallow(<SideBarItem icon='home' label='Menu Label' />);
    expect(wrapper).toMatchSnapshot();
  });
});