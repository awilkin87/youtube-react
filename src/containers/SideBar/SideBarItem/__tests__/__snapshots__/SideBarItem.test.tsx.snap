// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBarItem Renders SideBarItem w/ Highlight 1`] = `
<SideBarClickableItem
  className="sidebarItem highlight"
>
  <div
    className="alignmentContainer"
  >
    <span>
      <Icon
        as="i"
        name="home"
        size="large"
      />
    </span>
    <span>
      Menu Label
    </span>
  </div>
</SideBarClickableItem>
`;

exports[`SideBarItem Renders SideBarItem w/o Highlight 1`] = `
<SideBarClickableItem
  className="sidebarItem "
>
  <div
    className="alignmentContainer"
  >
    <span>
      <Icon
        as="i"
        name="home"
        size="large"
      />
    </span>
    <span>
      Menu Label
    </span>
  </div>
</SideBarClickableItem>
`;
